components/x-interaction/readme.md

```diff
@@ -108,3 +108,114 @@ export const BaseGreeting = ({greeting, actions}) => <div>
 
 export const Greeting = greetingActions(BaseGreeting);
 ```
+
+### Hydrating server-rendered markup
+
+When you have an `x-interaction` component rendered by the server, and you want to attach the client-side version of the component to handle the actions, rather than rendering the component manually (which might become unwieldy, especially if you have many components & instances on the page), you can have `x-interaction` manage it for you.
+
+There are two parts to this: serialising and hydrating.
+
+#### Serialising
+
+To ensure components are rendered with the same initial data on the client side, and keep track of all the instances that are rendered and their identifiers, you must create a serialiser on the server side. `x-interaction` exports a function `getInteractionSerialiser`. You *must* call this function *exactly once* at the start every HTTP request that responds with any number of `x-interaction` components.
+
+It returns a function that, when called, resets the serialisation state, and returns all serialisation data collected from any components that were rendered in the meantime, in the format of a `<script>` tag that should be included in the rendered page. This function *must* be called once at the end of every request when `getInteractionSerialiser`, _after_ all `x-interaction` components have been rendered. Any component rendered after this function is called won't have its data included in the output.
+
+If you call `getInteractionSerialiser` once for multiple requests, it will collect data for components that aren't rendered in the current request, causing inflated output size and losing that data for the pages that need it. Calling it multiple times per request is a no-op.
+
+If you don't call the function returned by `getInteractionSerialiser` at the end of the request, the data collected during this request will be lost, and possibly returned in a future request. When calling it multiple times, all calls after the first will return empty data, and including these in the response will potentially erase the global variable it outputs on the client side.
```

Comment:
Sounds like this data could be attached to the request object so that the server (of any flavour) can handle this lifecycle?

Reply:
i don't like the pattern of "just stick it on `req` or `res` because they're there", it's just a function that returns a string, there's nothing special about it, and i don't want to assume people are using express, or even using `http`, so i don't see any way of hooking it into the lifecycle?